src/common/components/game/GameArea.js

```diff
@@ -0,0 +1,282 @@
+import React, { Component } from 'react';
+import { arrayOf, bool, func, number, object, string } from 'prop-types';
+import Paper from 'material-ui/Paper';
+import screenfull from 'screenfull';
+
+import Chat from '../multiplayer/Chat';
+import { inBrowser } from '../../util/browser';
+
+import Board from './Board';
+import EndTurnButton from './EndTurnButton';
+import EventAnimation from './EventAnimation';
+import ForfeitButton from './ForfeitButton';
+import FullscreenMessage from './FullscreenMessage';
+import FullscreenToggle from './FullscreenToggle';
+import GameNotification from './GameNotification';
+import PlayerArea from './PlayerArea';
+import Sfx from './Sfx';
+import SoundToggle from './SoundToggle';
+import Status from './Status';
+import Timer from './Timer';
+import TutorialTooltip from './TutorialTooltip';
+import VictoryScreen from './VictoryScreen';
+
+/* Props shared by GameArea and GameAreaContainer. */
+export const gameProps = {
+  player: string,
+  currentTurn: string,
+  usernames: object,
+  winner: string,
+
+  selectedTile: string,
+  playingCardType: number,
+
+  status: object,
+  target: object,
+  attack: object,
+
+  blueHand: arrayOf(object),
+  orangeHand: arrayOf(object),
+
+  bluePieces: object,
+  orangePieces: object,
+
+  blueEnergy: object,
+  orangeEnergy: object,
+
+  blueDeck: arrayOf(object),
+  orangeDeck: arrayOf(object),
+
+  blueDiscardPile: arrayOf(object),
+  orangeDiscardPile: arrayOf(object),
+
+  eventQueue: arrayOf(object),
+  sfxQueue: arrayOf(string),
+  tutorialStep: object,
+
+  history: object,
+  location: object,
+  match: object,
+
+  gameOver: bool,
+  isTutorial: bool,
+  isMyTurn: bool,
+  isSpectator: bool,
+  isAttackHappening: bool,
+
+  actionLog: arrayOf(object),
+  socket: object
+};
+
+export default class GameArea extends Component {
+  /* eslint-disable react/no-unused-prop-types */
+  static propTypes = {
+    ...gameProps,
+
+    chatOpen: bool,
+    message: string,
+
+    onPassTurn: func,
+    onForfeit: func,
+    onTutorialStep: func,
+    onSendChatMessage: func,
+    onActivateObject: func,
+    onClickGameArea: func,
+    onClickEndGame: func,
+    onNextTutorialStep: func,
+    onPrevTutorialStep: func,
+    onSelectTile: func
+  };
+  /* eslint-enable react/no-unused-prop-types */
+
+  state = {
+    areaHeight: 1250,
+    boardSize: 1000,
+    chatOpen: true
+  };
+
+  componentDidMount() {
+    this.updateDimensions();
+    window.addEventListener('resize', () => { this.updateDimensions(); });
+  }
+
+  handleToggleFullScreen = () => {
+    screenfull.toggle(this.gameArea);
+  };
+
+  handleToggleChat = () => {
+    this.setState(state => ({ chatOpen: !state.chatOpen }));
+  };
+
+  loadBackground = () => inBrowser() ? require('../img/black_bg_lodyas.png') : '';
+
+  updateDimensions = () => {
+    const maxBoardHeight = window.innerHeight - 64 - 150;
+    const maxBoardWidth = window.innerWidth - 256;
+
+    this.setState({
```

Comment:
I'm still green to React, so I might be wrong on this. But this still seems like a muddying of concerns between the container and the component. From my reading the component should be fairly ignorant of React and the state and just be responsible for binding properties to elements and on handlers.

If my mental model is correct, the React-y thing to do here is pass in a reference to the `updateDimensions` function as defined in the container and then the component is only responsible for the invocation of that callback in the `componentDidMount` and the registration to the window resize event handler.

Reply:
I disagree. `GameArea` component ought to be responsible for all rendering-related work (while the container is mainly responsible for dispatch and routing), and I'd argue that everything in the `updateDimensions` function is rendering work.